### PR TITLE
fix: Incorrect depth readout after vertical re-scaling

### PR DESCRIPTION
**Pull request by nilscb**

### CODE REVIEW

File: typescript/packages/subsurface-viewer/src/layers/map/privateMapLayer.ts [hunk 1]
```diff
@@ -272,9 +272,15 @@ export default class PrivateMapLayer extends Layer<PrivateMapLayerProps> {
...
-                ? -info.coordinate[2]
-                : info.coordinate[2];
+            const zScale = this.props.modelMatrix
+                ? this.props.modelMatrix[10]
+                : 1;
+
+            const depth =
+                (this.props.ZIncreasingDownwards
+                    ? -info.coordinate[2]
+                    : info.coordinate[2]) / zScale;
```
Comment (hkfb) [inline typescript/packages/subsurface-viewer/src/layers/map/privateMapLayer.ts:282]: Can zScale be 0?

Comment (nilscb) [reply typescript/packages/subsurface-viewer/src/layers/map/privateMapLayer.ts:282]: No, but i can be arbitrarely small if one keeps on scaling down,

Comment (nilscb) [reply typescript/packages/subsurface-viewer/src/layers/map/privateMapLayer.ts:282]: I added a lower bound to it just to make sure.

**Review by t0oF-azpn — APPROVED** [review]

**Pull request merged**